features/environment.py

```diff
@@ -498,22 +499,47 @@ def after_scenario(context, scenario):
                 logging.warning("Failed to collect coverage")
 
 
-def _get_apparmor_logs_from_host():
-    # get apparmor DENIED messages from the host
-    with open("/var/log/syslog", "r") as syslog_fd:
-        syslog_messages = syslog_fd.readlines()
-        apparmor_denied = [
-            msg.strip()
-            for msg in syslog_messages
-            if ("DENIED" in msg and "ubuntu_pro_" in msg)
-        ]
-    return apparmor_denied
+def _get_relevant_apparmor_logs(context):
+    if hasattr(context, "machines") and SUT in context.machines:
+        sut = context.machines[SUT]
+        if sut.cloud == "lxd-container":
+            # get apparmor DENIED messages from the host
+            with open("/var/log/syslog", "r") as syslog_fd:
+                syslog_messages = syslog_fd.readlines()
+            apparmor_denied = [
+                msg.strip()
+                for msg in syslog_messages
+                if (
+                    "DENIED" in msg
+                    and "ubuntu_pro_" in msg
+                    and sut.instance.name in msg
+                )
+            ]
+            return apparmor_denied
+        else:
+            # get apparmor DENIED messages from the SUT
+            syslog_dest = os.path.join(
+                UA_TMP_DIR, "{}-syslog".format(sut.instance.name)
+            )
+            sut.instance.pull_file("/var/log/syslog", syslog_dest)
```

Comment:
The syslog file inside a lxd container unfortunately won't contain the apparmor denials. Nor anything else inside the container that I could find. It's always on the host in this case :(

Reply:
Ah, ok, in this case it's NOT a lxd container, so it's correct to grab it from the "inside".